common/web/nunjucks/README.md

```diff
@@ -0,0 +1,123 @@
+# @curiostack/nunjucks
+
+An extended version of [mozilla/nunjucks](https://github.com/mozilla/nunjucks),
+with some extra convenience methods added on.
+
+## Usage
+
+### Import
+
+The extended `nunjucks` object can be imported in other `curiostack` `yarn` modules
+using the package name `@curiostack/nunjucks`. It can work outside of curiostack as well,
+but your project must use typescript and be configured to compile this module.
+
+```typescript
+import nunjucks from '@curiostack/nunjucks'
+// no need for require('nunjucks')
+
+const env = nunjucks.Environment()
+// ...
+```
+
+### API
+
+The underlying API of `nunjucks` is unchanged, so you can
+[use the official docs](https://mozilla.github.io/nunjucks/getting-started.html) for reference.
+
+## Environment
+
+The [nunjucks Environment](https://mozilla.github.io/nunjucks/api.html#environment)
+class still functions as before. It just has been extended with some extra functionality.
+
+```typescript
+import nunjucks from '@curiostack/nunjucks'
+
+const env = nunjucks.Environment()
+// ...
+
+const result = env.render('template.njk')
+// `result` is rendered with all of the base `nunjucks.Environment` logic
+//   as well as the extended `curiostack` `nunjucks.Environment` functionality
+```
+
+> This means that you can NOT access the extended features, if you use the
+> [simple nunjucks API](https://mozilla.github.io/nunjucks/api.html#simple-api),
+> without initializing the extended `nunjucks.Environment`
+
+### Filters
+
+The `Environment` object is extended with some extra convenience filters.
+
+#### allAfterLine(lineMatcher: string, [doInclusionCheck: boolean])
+
+Scans the content of a multiline text and returns all the text after a line
+matching a `regex` expression. Only exact matches will count.
+
+Alternatively, if `doInclusionCheck` is `true`, the line is checked if it
+includes `lineMatcher` within it, instead of doing a `regex` evaluation.
+
+##### input
+
+```
+{{ 'foo\n_@@_\nbar\nbaz' | allAfterLine('.*@@.*') }}
```

Comment:
Ah sorry realized this isn't what I was suggesting - it's definitely one option, but we should predict what's most common. I guess it's either inclusion check or exact match, and I suspect it's inclusion check. In which case, the default can be regex, but a `find` regex instead of `exact match` regex.

```
--- cats and dogs ---
cats meow
dogs bark
--- cats and dogs ---
```

`allBetweenLines('cats and dogs')` <- matches
`allBetweenLines('.*cats and dogs.*')` <- matches
`allBetweenLines('^cats and dogs$')` <- doesn't match
`allBetweenLines('cats and dogs', exactMatch: true)` <- doesn't match
`allBetweenLines('--- cats and dogs ---', exactMatch: true)` <- matches
`allBetweenLines('^--- cats and dogs ---$')` <- matches

Does this feel like what we'd expect from common usage?

Another option is strings are exact match, regex are regex in `find` mode, which would be the same as inclusion. Then no flag

`allBetweenLines('cats and dogs')` <- doesn't match
`allBetweenLines('.*cats and dogs.*')` <- doesn't match
`allBetweenLines(/cats and dogs/)` <- matches
`allBetweenLines(/.*cats and dogs;*/)` <- matches
`allBetweenLines('--- cats and dogs ---')` <- matches

The parameter can accept `string | Regexp` and wrap a passed in string with `Regexp(^arg$)`. What do you think?

Reply:
The second approach (`string | Regexp`) is what I tried as the very first thing, before even submitting a PR for this `nunjucks` change. But the `nunjucks` template engine doesn't support it, when parsing templates.
```
{% filter allBetweenLines(/cats and dogs/) %}{% include('catdog.txt') %}{% endfilter %}
// nunjucks doesn't understand this one ^
{% filter allBetweenLines(new RegExp('cats and dogs')) %}{% include('catdog.txt') %}{% endfilter %}
// nunjucks doesn't understand this one either ^
{% filter allBetweenLines('cats and dogs', true) %}{% include('catdog.txt') %}{% endfilter %}
// this one works obviously ^
```

I'm liking the first approach though. I think it is more intuitive, with the `find` mode as default and an optional `exactMatch`. And then there is no confusion whether it is a `regex` check or not (always will be `regex`), or whether it is exact or not.
:+1: 

With the current code it's super easy to code in this behavior.

Reply:
Ah thanks for the explanation, then yeah let's go with the find-by-default and string-only